tests/integration_test.rs

```diff
@@ -117,6 +117,34 @@ fn test_long_sleep() {
     assert!(!traces[0].active);
 }
 
+#[test]
+fn test_thread_names() {
+    #[cfg(target_os = "macos")]
+    {
+        // We need root permissions here to run this on OSX
+        if unsafe { libc::geteuid() } != 0 {
+            return;
+        }
+    }
+    let config = Config {
+        include_idle: true,
+        ..Default::default()
+    };
+    let mut runner = TestRunner::new(config, "./tests/scripts/thread_names.py");
+
+    let traces = runner.spy.get_stack_traces().unwrap();
+    assert_eq!(traces.len(), 11);
+
+    let mut expected_threads: HashSet<String> =
+        (0..10).map(|n| format!("CustomThreadName-{}", n)).collect();
+    expected_threads.insert("MainThread".to_string());
+    let detected_threads: HashSet<String> = traces
+        .iter()
+        .map(|trace| trace.thread_name.as_ref().unwrap().clone())
```

Comment:
The unwrap on this line is causing issues on travis with the x86_64 linux variants:  https://travis-ci.org/github/benfred/py-spy/builds/684995442 .  I believe this is probably because we are using a python2 interpreter there - and we can only get thread names from version 3.6 on.

Can you only tun this on python 3.6+ ? There is code in test_local_vars that does this